### PR TITLE
Presentation Blaster 2.0

### PR DESCRIPTION
**Pull request by aeslaughter**
This adds a tool for building markdown presentations, using MOOSE flavored markdown, that used [Reveal.js](http://lab.hakim.se/reveal-js/#/) as the engine. There is about 100 lines for this, the old blaster has about a 100 files.

To actually view the presentation you need to have "reveal.js" installed, so I added it as a sub-module. I am not sure this is the solution we want, so I am putting this up for discussion.

If you have your environment setup for building docs, then simply init the reveal.js submodule  and you should be able to build the demo presentation.

```
cd ~/projects/moose/docs
git submodule update --init
./mooseslides.py slides/demo.md
open slides/demo.html
```

This also does some clean-up on the extension constructors.

(refs #6699)

### CODE REVIEW

**Comment by aeslaughter** [comment]
Another solution, which is probably better, is to have reveal.js installed in the package: /opt/moose/reveal.js

**Comment by aeslaughter** [comment]
I removed submodule, that was silly.

```
cd ~/projects
git clone git@github.com:hakimel/reveal.js.git
cd ~/projects/moose/docs
export MOOSE_REVEAL_JS_DIR=~/projects/reveal.js/
./mooseslides.py slides/demo.md
open slides/demo.html
```

File: docs/content/help/contact_us.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+* MOOSE Users for technical Q&A (moderate traffic): [moose-users@googlegroups.com](https://groups.google.com/forum/#!forum/moose-users)
+* MOOSE Announcements (very light traffic): [moose-announce@googlegroups.com](https://groups.google.com/forum/#!forum/moose-announce)
+
+!!! note "Gmail Users"
+    Just click the "Join group" button
+
+!!! note "Non-Gmail Users"
+    Send an email to:
+    * [moose-users+subscribe@googlegroups.com](mailto:moose-users+subscribe@googlegroups.com) and/or
+    * [moose-announce+subscribe@googlegroups.com](mailto:moose-announce+subscribe@googlegroups.com)
```
Comment (milljm) [inline docs/content/help/contact_us.md]: The default theme is hard to read. Its dark blue links on top of a light blue admonition. I know there is no way around this currently however. Just pointing it out. Its even worse when you use dorkmode ;)

File: docs/pages.yml [hunk 2]
```diff
@@ -10,15 +10,19 @@
...
+        - Systems: !include content/modules/phase_field/systems/pages.moose.yml
+    - Contact:
+        - Systems: !include content/modules/contact/systems/pages.moose.yml
+- Utilites:
+    - Documentation:
+        - Creation: 'content/utilities/documentation/create.md'
+        - Moose Flavored Markdown: 'content/utilities/documentation/moose_flavored_markdown.md'
+- Help:
+    - Frequently Asked Questions: 'content/help/faq.md'
+    - Contact Us: 'content/help/contact_us.md'
```
Comment (milljm) [inline docs/pages.yml]: When serving with 'themes', we start to collide with our Nav bar:

screen shot 2016-08-26 at 10 54 23 am


Looks like we are going to need to trim back a few things.

File: docs/content/help/contact_us.md (hunk 1 above)
Comment (aeslaughter) [inline docs/content/help/contact_us.md]: I would like to change our admonition to the readthedoc theme style, I think we just need to copy paste some stuff from that theme into css/moose.css. Run with --theme readthedocs and you will see what I am after.

File: docs/pages.yml (hunk 2 above)
Comment (aeslaughter) [inline docs/pages.yml]: What if rename "Application Development" to something shorter: "App Development"

**Comment by milljm** [comment]
👍

**Comment by jwpeterson** [comment]
How does this affect (if at all) the MOOSE training workshop slides?

**Comment by aeslaughter** [comment]
This will not impact the current system, the idea is that the current wiki
content needs to be moved into the repo and setup to use the new system.

On Mon, Aug 29, 2016 at 10:02 AM, John W. Peterson <notifications@github.com

**Pull request merged**